ADR/XXXX-pipeline-caching-feature-flag.md

```diff
@@ -0,0 +1,79 @@
+# XXXX. Pipeline Caching Feature Flag Configuration
+
+Date: 2025-10-23
+
+## Status
+
+Proposed
+
+## Context
+
+Following the implementation of container build layer caching via Squid HTTP proxies as described in [ADR 0047](0047-caching-for-container-build-layers.md), we need to provide users with flexible control over when and how the proxy caching is enabled for their pipelines. Users need the ability to:
+
+* Enable or disable HTTP proxy use for individual pipelines without knowing technical details about the proxy configuration
+* Control proxy usage at the namespace level for all pipelines running in that namespace
+* Control proxy usage at the cluster level for all pipelines in the cluster
+* Apply GitOps processes for this configuration, similar to other Konflux configuration elements
+
+The current design requires users to manually configure `HTTP_PROXY` and `NO_PROXY` parameters to the buildah task, which exposes implementation details and creates a poor user experience. We need a more user-friendly configuration mechanism that abstracts away the technical details while providing appropriate levels of control.
+
+## Decision
+
+We will implement a hierarchical feature flag system for controlling pipeline caching proxy usage with the following configuration levels and precedence:
+
+### Configuration Levels
+
+1. **Pipeline Level**: An `enable-cache-proxy` parameter that can be set in individual pipeline definitions
+2. **Namespace Level**: An `enable-cache-proxy` value in the `konflux-config` config map in the pipeline's namespace
+3. **Cluster Level**: An `enable-cache-proxy` value in the `cluster-config` config map in the `konflux-info` namespace
```

Comment:
Are those `*-config` Config Maps introduced here or they already exist?

Reply:
How init tasks will get access to those config maps?

Will we update PLR to mount configmaps from tenant?

And the global configmap, we cannot mount it, are we going to add access to each SA to fetch it via k8s API?